packages/core/src/types.ts

```diff
@@ -202,6 +209,11 @@ export interface RuleMeta {
    * @private
    */
   __hash?: string
+
+  /**
+   * Depend on preflight keys
+   */
+  preflightKeys?: Arrayable<string>
```

Comment:
I am a bit worried that if this would be too coupled with our implementation of preset-mini (meaning it might not be that useful for other custom presets)

I wonder if we could put this in rule's meta, and instead of passing a `keys`, we pass a list of "Active rules"? So `preflightKeys` can be an internal convention we use for `preset-mini`, while it's also possible to pass more information other than just keys. WDYT?

Reply:
There is some coupling～😅, I will continue to explore new implementation methods.